collect_app/src/main/java/org/odk/collect/android/tasks/sms/SmsService.java

```diff
@@ -0,0 +1,379 @@
+import com.evernote.android.job.util.support.PersistableBundleCompat;
+
+import org.odk.collect.android.dao.FormsDao;
+import org.odk.collect.android.dao.InstancesDao;
+import org.odk.collect.android.events.RxEventBus;
+import org.odk.collect.android.events.SmsEvent;
+import org.odk.collect.android.jobs.SmsSenderJob;
+import org.odk.collect.android.preferences.GeneralSharedPreferences;
+import org.odk.collect.android.preferences.PreferenceKeys;
+import org.odk.collect.android.provider.InstanceProviderAPI;
+import org.odk.collect.android.tasks.sms.contracts.SmsSubmissionManagerContract;
+import org.odk.collect.android.tasks.sms.models.Message;
+import org.odk.collect.android.tasks.sms.models.MessageStatus;
+import org.odk.collect.android.tasks.sms.models.SentMessageResult;
+import org.odk.collect.android.tasks.sms.models.SmsSubmission;
+import org.odk.collect.android.tasks.sms.models.SubmitFormModel;
+import org.odk.collect.android.utilities.ArrayUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+
+import javax.inject.Inject;
+
+import timber.log.Timber;
+
+import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.AUTO_DELETE;
+import static org.odk.collect.android.tasks.sms.Mapper.toMessageStatus;
+import static org.odk.collect.android.tasks.sms.SmsUtils.checkIfSentIntentExists;
+import static org.odk.collect.android.tasks.sms.SmsUtils.checkIfSubmissionSentOrSending;
+import static org.odk.collect.android.tasks.sms.SmsUtils.getDisplaySubtext;
+import static org.odk.collect.android.utilities.FileUtil.getFileContents;
+import static org.odk.collect.android.utilities.FileUtil.getSmsInstancePath;
+
+/**
+ * Core class that contains all the business logic and services that are utilized to send,track
+ * and store Form SMS data.
+ */
+public class SmsService {
+
+    private final SmsManager smsManager;
+    private final SmsSubmissionManagerContract smsSubmissionManager;
+    private final Context context;
+    private final RxEventBus rxEventBus;
+    private final InstancesDao instancesDao;
+    private final FormsDao formsDao;
+
+    @Inject
+    public SmsService(SmsManager smsManager, SmsSubmissionManagerContract smsSubmissionManager, InstancesDao instancesDao, Context context, RxEventBus rxEventBus, FormsDao formsDao) {
+        this.smsManager = smsManager;
+        this.smsSubmissionManager = smsSubmissionManager;
+        this.context = context;
+        this.instancesDao = instancesDao;
+        this.rxEventBus = rxEventBus;
+        this.formsDao = formsDao;
+    }
+
+    public void submitForms(long[] instanceIds) {
+
+        Long[] instanceIdObjects = ArrayUtils.toObject(instanceIds);
+        List<Long> list = java.util.Arrays.asList(instanceIdObjects);
+
+        StringBuilder selection = new StringBuilder();
+
+        Iterator<Long> it = list.iterator();
```

Comment:
Okay I was using this logic and ended up having some weird crashes as it wasn't checking a particular case. I modified it to do so but still was having a crash due an SQLiteException that was being thrown when an additional "or" is being attached to the statement so I just reverted until I figure it out since I wanted to remove some other things.